typescript-definitions-derive/src/tests.rs

```diff
@@ -3,108 +3,285 @@ use super::Typescriptify;
 
 #[cfg(test)]
 mod macro_test {
-    // use proc_macro2::Ident;
     use super::Typescriptify;
-    use insta::assert_snapshot_matches;
     use quote::quote;
+
+    macro_rules! assert_conversion {
+        ($tokens:expr,$expected:expr) => {{
+            let declarations = Typescriptify::new($tokens)
+                .parse()
+                .export_type_definition_source()
+                .declarations;
+
+            assert_eq!(declarations, $expected)
+        }};
+    }
+
+    // The crate only converts complex types, so we need to wrap the type in a struct to check its
+    // conversion. We could also write these tests at the conversion logic inside the crate but
+    // right now I'm just tring to get as much value for as little effort as possible.
+    macro_rules! assert_type_conversion {
+        ($rust:ty,$ts:tt) => {
+            assert_conversion!(
+                struct_with_type!($rust),
```

Comment:
Maybe we could drop `quote~` in here directly, but I was kind of ok giving it a name.